tasks/Task.cpp

```diff
@@ -7,29 +7,29 @@ using namespace vizkit3d_world;
 const std::string Task::JOINTS_CMD_POSFIX = ":joints_cmd";
 
 Task::Task(std::string const& name) :
-        TaskBase(name), vizkit3dWorld(NULL)
+        TaskBase(name)
 {
 }
 
 Task::Task(std::string const& name, RTT::ExecutionEngine* engine) :
-        TaskBase(name, engine),
-        vizkit3dWorld(NULL)
+        TaskBase(name, engine)
 {
 }
 
 Task::~Task() {
 }
 
-void Task::setupJointsPorts() {
-
-    if (vizkit3dWorld) {
+void Task::setupJointsPorts() 
+{
+    if(vizkit3dWorlds[0])
```

Comment:
`vizkit3dWorlds.empty()` or you access uninitialized memory if the vector is empty

While you are at it, change the `if ( ) { BIG } else { return }` into early return (`if (! ...) return;`)